client/packages/programs/src/JsonForms/common/components/Array/common.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { useMemo, useState } from 'react';
+import {
+  ArrayControlProps,
+  findUISchema,
+  ControlElement,
+  composePaths,
+  createDefaultValue,
+  JsonSchema,
+} from '@jsonforms/core';
+import { JsonFormsDispatch } from '@jsonforms/react';
+import {
+  Box,
+  Typography,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+} from '@mui/material';
+import {
+  IconButton,
+  PlusCircleIcon,
+  MinusCircleIcon,
+  ChevronDownIcon,
+  useTranslation,
+  ConfirmationModal,
+} from '@openmsupply-client/common';
+import { RegexUtils } from '@common/utils';
+import { z, ZodSchema } from 'zod';
+import { useZodOptionsValidation } from '../../hooks/useZodOptionsValidation';
+import {
+  FORM_LABEL_COLUMN_WIDTH,
+  FORM_INPUT_COLUMN_WIDTH,
+} from '../../styleConstants';
+import { JsonData } from '../../JsonForm';
+import { EnumArrayComponent } from './';
+
+export interface UISchemaWithCustomProps extends ControlElement {
+  defaultNewItem?: JsonData;
+  itemLabel?: string;
+}
+
+export interface ArrayControlCustomProps extends ArrayControlProps {
+  uischema: UISchemaWithCustomProps;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  data: JsonData[] | undefined;
+  isElementEditable?: (child: JsonData, index: number) => boolean;
+  getItemLabel?: (
+    child: JsonData,
+    index: number,
+    isExpanded: boolean
+  ) => JSX.Element;
+  zOptions?: ZodSchema;
+}
+
+export const CommonOptions = z
+  .object({
+    detail: z.object({ type: z.string(), elements: z.array(z.any()) }),
+    /**
+     * If true, display items in reverse order (newest first)
+     */
+    reverse: z.boolean().optional(),
+    /**
+     * If true, all elements will be expanded (Accordions) on load. Otherwise,
+     * they'll start closed and only new items will start expanded
+     */
+    defaultExpanded: z.boolean().optional(),
+    /**
+     * Restrictions for which elements can be edited
+     */
+  })
+  .strict();
+
+const isStringEnum = (
+  schema: JsonSchema,
+  _data: JsonData[]
+): _data is string[] => {
+  return !!schema.enum && schema.type === 'string';
+};
+
+export const ArrayCommonComponent = (props: ArrayControlCustomProps) => {
```

Comment:
Exactly the same as the previous "Array" component, but with capability of overriding certain default (itemLabel, editablity, Options) via props